pdf.go

```diff
@@ -859,6 +859,45 @@ func (w *pdfPageWriter) EndTextObject() {
 	w.inTextObject = false
 }
 
+/* Extract from the pdf specification:
+When a string is written by enclosing the data in parentheses, bytes whose values are the
+same as thoseof  the  ASCII  characters  left  parenthesis  (40),  right  parenthesis  (41),
+and  back-slash  (92)  must  be  preceded  by  a  backslash  character. (...)
+These  rules  apply  to  each individual  byte  in  a  string  object, whether  the  string
+is  interpreted  by  the  text-showing operators as single-byte or multiple-byte character codes.
+*/
+type stringBytesEscaper struct {
+	w           io.Writer
+	escapeChars []byte
+}
+
+func (r stringBytesEscaper) Write(p []byte) (n int, err error) {
+	var k int
+	i := 0
+	for j, b := range p {
+		if b != '(' && b != ')' && b != '\\' {
+			continue
+		}
+		if i != j {
+			k, err = r.w.Write(p[i:j])
+			n += k
+			if err != nil {
+				return
+			}
+		}
+		_, err = r.w.Write(r.escapeChars)
```

Comment:
You don't update `n` here?

Reply:
No, because otherwise from the caller's view, more bytes gets written as asked (and the test panic with `invalid WriteString count`).

Reply:
You are right, `io.Writer` can at most return `len(p)` https://golang.org/pkg/io/#Writer